src/utils/memoize.js

```diff
@@ -0,0 +1,52 @@
+import fastMemoize from 'fast-memoize'
+
+import { MEMOIZE_CACHE_CLEAR_THRESHOLD } from '../constants'
+
+// This is basically the same as fast-memoize's default cache, except that it
+// clears the cache once the size exceeds MEMOIZE_CACHE_CLEAR_THRESHOLD to
+// prevent memory bloat.
+// https://github.com/caiogondim/fast-memoize.js/blob/5cdfc8dde23d86b16e0104bae1b04cd447b98c63/src/index.js#L114-L128
+/**
+ * @ignore
+ */
+export class MemoizeCache {
+  cache = {}
+
+  /**
+   * @param {Object} [config] Can also contain the config options used to
+   * configure fast-memoize.
+   * @param {number} [config.cacheSize]
+   * @see https://github.com/caiogondim/fast-memoize.js
+   */
+  constructor({ cacheSize = MEMOIZE_CACHE_CLEAR_THRESHOLD } = {}) {
+    this.cacheSize = cacheSize
+  }
+
+  has(key) {
+    return key in this.cache
+  }
+
+  get(key) {
+    return this.cache[key]
+  }
+
+  set(key, value) {
+    if (Object.keys(this.cache).length > this.cacheSize) {
+      this.cache = {}
+    }
```

Comment:
If you stored your cache entries as an array and lookup via filter, you could bump values to the "top" of the cache as you read them so that when it does come time to make room, you could just bump off enough entries from the back half of the array to get under your threshold without needing to clear the whole thing, potentially reducing "lag spikes" as the whole cache re-hydrates! It might be a minimal performance hit of a few cycles on lookup, but might have more of the intended effect! Also, consider making the cache size a kb metric rather than a keylength one!

Reply:
This is an interesting idea! 

In the interest of avoiding premature optimization, I think it would be best to first identify a scenario where the current caching strategy negatively affects the player's experience in a perceptible way. I haven't personally noticed that, but that doesn't mean nobody else does.

It's not clear that the proposed alternative makes sense for all scenarios because of the `O(n)` lookup that would be introduced (it's currently `O(1)`). However, a variety of caching approaches can coexist in the code base and one or the other could be used depending on the use case.

@thatsanicehat is this an idea you'd like to explore more? If so I can make a ticket for you. I'm not personally seeing the need for the change, but I'd be happy to work with you on a PR. Let me know!

------------------------------

For context, this code was moved from [`utils/index.js`](https://github.com/jeremyckahn/farmhand/blob/eea169a886501d95f3633d571bee70dbc69e0b16/src/utils/index.js#L125-L153) rather than added added in this PR, so I'd want to save any substantive changes to it for another PR.